--- build_defs.bzl.1	2023-10-06 11:21:14.220307188 +0200
+++ build_defs.bzl	2023-10-06 11:24:35.624577060 +0200
@@ -24,7 +24,9 @@
 
 # Builds a Python extension module using pybind11.
 # This can be directly used in python with the import statement.
-# This adds rules for a .so binary file.
+# This adds rules for .so and .pyd binary files, as well as
+# a base target that selects between them depending on the platform
+# (.pyd for windows, .so otherwise).
 def pybind_extension(
         name,
         copts = [],
@@ -40,16 +42,29 @@
         name = name + ".so",
         copts = copts + PYBIND_COPTS + ["-fvisibility=hidden"],
         features = features + PYBIND_FEATURES,
-        linkopts = linkopts + select({
-            "@pybind11//:osx": [],
-            "//conditions:default": ["-Wl,-Bsymbolic"],
-        }),
+        linkopts = linkopts + ["-Wl,-Bsymbolic"],
         linkshared = 1,
         tags = tags,
         deps = deps + PYBIND_DEPS,
         **kwargs
     )
 
+    native.genrule(
+        name = name + "_pyd",
+        srcs = [name + ".so"],
+        outs = [name + ".pyd"],
+        cmd = "cp $< $@",
+    )
+
+    native.py_library(
+        name = name,
+        data = select({
+            "@platforms//os:windows": [":" + name + ".pyd"],
+            "//conditions:default": [":" + name + ".so"],
+        }),
+    )
+
+
 # Builds a pybind11 compatible library. This can be linked to a pybind_extension.
 def pybind_library(
         name,
